agent/consul/acl_test.go

```diff
@@ -1264,6 +1264,35 @@ func TestACLResolver_DownPolicy(t *testing.T) {
 		requireIdentityCached(t, r, tokenSecretCacheID("found"), false, "no longer cached")
 	})
 
+	t.Run("Cache-Error", func(t *testing.T) {
+		_, rawToken, _ := testIdentityForToken("found")
+		foundToken := rawToken.(*structs.ACLToken)
+		secretID := foundToken.SecretID
+
+		delegate := &ACLResolverTestDelegate{
+			enabled:       true,
+			datacenter:    "dc1",
+			legacy:        false,
+			localTokens:   false,
+			localPolicies: false,
+			tokenReadFn: func(_ *structs.ACLTokenGetRequest, reply *structs.ACLTokenResponse) error {
+				return acl.ErrPermissionDenied
```

Comment:
slight nit but can we return some `testErr := fmt.Errorf("network error")`?

The type of error returned here isn't too important but I think using an error from the `acl` package (and the fact that we have special handling for a bunch of ACL error types) makes it a bit confusing.